### PR TITLE
Run CPO conformance test in multinode Kubernetes cluster

### PR DESCRIPTION
**Pull request by lingxiankong**
Fixes https://github.com/kubernetes/kubernetes/issues/91739

### CODE REVIEW

**Comment by lingxiankong** [comment]
@ramineni @chrigl could you please have a look at this PR? I've set up a local environment using the proposed ansible playbook, but the kubetest command failed with random test cases.

File: playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance/run.yaml [hunk 1]
```diff
@@ -1,157 +1,146 @@
...
-          # temproary change to solve memory issue
-          make WHAT="test/e2e/e2e.test vendor/github.com/onsi/ginkgo/ginkgo cmd/kubectl"
+- hosts: k8s-node
+  name: join kubernetes nodes
+  become: yes
+  roles:
+    - role: create-multinodes-k8s-cluster-with-kubeadm
+      k8s_role_to_deploy:
+        - node
+      kubeadm_join_cmd: "{{ hostvars['k8s-master']['kubeadm_join_cmd']['stdout'] }}"
```
Comment (oomichi) [inline playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance/run.yaml:105]: Don't we need to wait for complete node joining?
Basically it takes time to become "Ready" state on worker node after operating `kubeadm join` command.

File: playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance/run.yaml [hunk 2]
```diff
@@ -1,157 +1,146 @@
...
+            extraArgs:
+              "cloud-provider": "external"
+          ---
+          apiVersion: kubeadm.k8s.io/v1beta2
+          kind: InitConfiguration
+          nodeRegistration:
+            kubeletExtraArgs:
+              cloud-provider: "external"
+
   roles:
```
Comment (oomichi) [inline playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance/run.yaml:33]: Quick test result:
```
ERROR! the role 'config-golang' was not found in /home/ubuntu/openlab-zuul-jobs/playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance/roles:/home/ubuntu/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/home/ubuntu/openlab-zuul-jobs/playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance

The error appears to be in '/home/ubuntu/openlab-zuul-jobs/playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance/run.yaml': line 34, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  roles:
    - config-golang
      ^ here
```

Comment (lingxiankong) [reply playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance/run.yaml:33]: Thanks for the testing, you can find the roles used in this playbook here https://github.com/theopenlab/openlab-zuul-jobs/tree/master/roles, you need to copy the role folder to your ansible role path.

File: playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance/run.yaml [hunk 3]
```diff
@@ -1,157 +1,146 @@
...
 
-          # Create cloud-config
-          mkdir -p /etc/kubernetes/
-          cat << EOF >> /etc/kubernetes/cloud-config
+          export ARCH=amd64
+          export IMAGE_NAMES=openstack-cloud-controller-manager
+          export VERSION=latest
+          make build-images
+
+          docker tag k8scloudprovider/openstack-cloud-controller-manager-amd64:latest k8scloudprovider/openstack-cloud-controller-manager:latest
```
[the remark this answers is not in the document]
Comment (lingxiankong) [reply playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance/run.yaml]: Done

File: playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance/run.yaml (hunk 1 above)
Comment (lingxiankong) [reply playbooks/cloud-provider-openstack-acceptance-test-e2e-conformance/run.yaml:105]: Done

**Pull request merged**